Mathlib/RingTheory/Ideal/Pointwise.lean

```diff
@@ -28,26 +28,34 @@ namespace Ideal
 
 section Monoid
 
-variable [Monoid M] [CommRing R] [MulSemiringAction M R]
+variable [Monoid M] [Semiring R] [MulSemiringAction M R]
 
 /-- The action on an ideal corresponding to applying the action to every element.
 
 This is available as an instance in the `Pointwise` locale. -/
-protected def pointwiseMulSemiringAction : MulSemiringAction M (Ideal R) where
+protected def pointwiseMulAction : MulAction M (Ideal R) where
   smul a := Ideal.map (MulSemiringAction.toRingHom _ _ a)
   one_smul I :=
     congr_arg (I.map ·) (RingHom.ext <| one_smul M) |>.trans I.map_id
   mul_smul _ _ I :=
     congr_arg (I.map ·) (RingHom.ext <| mul_smul _ _) |>.trans (I.map_map _ _).symm
+
+scoped[Pointwise] attribute [instance] Ideal.pointwiseMulAction
+
+open Pointwise
+
+/-- The action on an ideal corresponding to applying the action to every element.
+
+This is available as an instance in the `Pointwise` locale. -/
+protected def pointwiseMulSemiringAction {R : Type*} [CommRing R] [MulSemiringAction M R] :
+    MulSemiringAction M (Ideal R) where
   smul_one a := by simp only [Ideal.one_eq_top]; exact Ideal.map_top _
   smul_mul a I J := Ideal.map_mul (MulSemiringAction.toRingHom _ _ a) I J
   smul_add _ I J := Ideal.map_sup _ I J
```

Comment:
This holds in the non-commutative case, right? Can you add a `DistribMulAction M (Ideal R)` instance as well?

Reply:
Ah nevermind, I think you can just change the `MulAction` above to a `DistribMulAction`

Reply:
Thanks!